src/test/pull_hf_model_test.cpp

```diff
@@ -0,0 +1,149 @@
+        server.setShutdownRequest(0);
+        // Clone sets readonly - need to remove it before we can delete on windows
+        RemoveReadonlyFileAttributeFromDir(this->directoryPath);
+        TestWithTempDir::TearDown();
+    }
+};
+
+TEST_F(HfDownloaderPullHfModel, PositiveDownload) {
+    std::string modelName = "OpenVINO/Phi-3-mini-FastDraft-50M-int8-ov";
+    std::string downloadPath = ovms::FileSystem::appendSlash(this->directoryPath) + "repository";
+    this->ServerPullHfModel(modelName, downloadPath);
+    std::string modelPath = ovms::FileSystem::appendSlash(downloadPath) + "openvino_model.bin";
+    ASSERT_EQ(std::filesystem::exists(modelPath), true);
+    ASSERT_EQ(std::filesystem::file_size(modelPath), 52417240);
+}
+
+TEST(HfDownloaderClassTest, Constructor) {
+    std::string modelName = "model/name";
+    std::string downloadPath = "/path/to/Download";
+    std::string hfEndpoint = "www.new_hf.com";
+    std::string hfToken = "123$$o_O123!AAbb";
+    std::string httpProxy = "https://proxy_test1:123";
+    std::unique_ptr<ovms::HfDownloader> hfDownloader = std::make_unique<ovms::HfDownloader>(modelName, downloadPath, true, hfEndpoint, hfToken, httpProxy);
+    ASSERT_EQ(hfDownloader->isPullHfModelModeOn(), true);
+}
+
+class TestHfDownloader : public ovms::HfDownloader {
+public:
+    TestHfDownloader(const std::string& sourceModel, const std::string& downloadPath, bool pullHfModelMode, const std::string& hfEndpoint, const std::string& hfToken, const std::string& httpProxy) :
+        HfDownloader(sourceModel, downloadPath, false, hfEndpoint, hfToken, httpProxy) {}
+    std::string GetRepoUrl() { return HfDownloader::GetRepoUrl(); }
+    std::string GetRepositoryUrlWithPassword() { return HfDownloader::GetRepositoryUrlWithPassword(); }
+    bool CheckIfProxySet() { return HfDownloader::CheckIfProxySet(); }
+    void setProxy(const std::string& proxy) { this->httpProxy = proxy; }
+    void setEndpoint(const std::string& endpoint) { this->hfEndpoint = endpoint; }
+    const std::string& getEndpoint() { return this->hfEndpoint; }
+    const std::string& getProxy() { return this->httpProxy; }
+};
+
+TEST(HfDownloaderClassTest, Methods) {
+    std::string modelName = "model/name";
+    std::string downloadPath = "/path/to/Download";
+    std::string hfEndpoint = "www.new_hf.com/";
+    std::string hfToken = "123$$o_O123!AAbb";
+    std::string httpProxy = "https://proxy_test1:123";
+    std::unique_ptr<TestHfDownloader> hfDownloader = std::make_unique<TestHfDownloader>(modelName, downloadPath, false, hfEndpoint, hfToken, httpProxy);
+    ASSERT_EQ(hfDownloader->isPullHfModelModeOn(), false);
+    ASSERT_EQ(hfDownloader->getProxy(), httpProxy);
+    ASSERT_EQ(hfDownloader->CheckIfProxySet(), true);
+
+    hfDownloader->setProxy("");
+    ASSERT_EQ(hfDownloader->CheckIfProxySet(), false);
+    ASSERT_EQ(hfDownloader->getEndpoint(), "www.new_hf.com/");
+    ASSERT_EQ(hfDownloader->GetRepoUrl(), "https://www.new_hf.com/model/name");
```

Comment:
Those are verified in GetRepositoryUrlWithPassword. We don't need to test if setting those values in constructor works.

Only checking for proxy makes sense but even then it is better to document what is considered proxy not set for hf downloader with 
EXPECT_EQ(TestHfDownloader(modelName, downloadPath, false, hfEndpoint, hfToken, "").isProxySet(), false)
or
EXPECT_EQ(TestHfDownloader(modelName, downloadPath, false, hfEndpoint, hfToken, "").isProxySet(), false)

Reply:
This is standard unit test approach to test all methods of a class. Specifically when one method returns values that are used in another method. I can change the EXPECT_EQ for proxy though for better readability.